functorch/functorch/experimental/cond.py

```diff
@@ -0,0 +1,137 @@
+import torch
+from torch._C import DispatchKey, DispatchKeySet, ExcludeDispatchKeyGuard
+from functorch.experimental.ops import PyOperator
+from torch.utils._pytree import tree_flatten
+from torch.fx.experimental.proxy_tensor import get_isolated_graphmodule, disable_proxy_modes_tracing
+import torch.utils._pytree as pytree
+from torch.dispatch.dispatcher import dispatcher_singleton
+
+"""
+We're going to define a `cond` operation.
+In order to do this, we need implementations for each of the dispatch keys.
+"""
+from contextlib import contextmanager
+
+# TODO(voz): Move out somewhere else once other py dispatched ops need it
+@contextmanager
+def suspend_mode(mode):
+    assert(mode is not None), "Cannot suspend None mode"
+    torch._C._set_torch_dispatch_mode(None)
+    try:
+        yield
+    finally:
+        torch._C._set_torch_dispatch_mode(mode)
+
+@contextmanager
+def enable_mode(mode):
+    curr_mode = torch._C._get_torch_dispatch_mode()
+    torch._C._set_torch_dispatch_mode(mode)
+    try:
+        yield
+    finally:
+        torch._C._set_torch_dispatch_mode(curr_mode)
+
+
+def trace_cond(proxy_mode, func_overload, args, kwargs=None):
+    assert kwargs is None or not kwargs
+    pred, true_fn, false_fn, operands = args
+
+    def _unwrap_proxy(e):
+        if hasattr(e, "__dict__") and proxy_mode.tracer in e.__dict__:
+            proxy_out = e.__dict__[proxy_mode.tracer]
```

Comment:
The most recent https://github.com/pytorch/pytorch/pull/83646 adds helper functions you can use to access this, which is good because I changed the API oops

Reply:
hah sounds good, will use, this was hacked on top of your PR before it landed.